eox_tagging/api/v1/filters.py

```diff
@@ -26,20 +29,42 @@ class Meta:
         model = Tag
         fields = ["key", "status", "tag_type", "tag_value"]
 
+    @staticmethod
+    def _map_to_model(name, value):
+        """
+        Used to map filter names to model names and their types.
+
+        Example usage:
+            >>> model_field_name, value = self._map_to_model("certificate_id", "1")
+            >>> model_field_name
+            >>> 'id'
+            >>> value
+            >>> 1
+        For now, it just maps certificates.
+        """
+        if name.startswith("certificate_"):
+            name = name[len("certificate_"):]
+            value = int(value) if name == "id" else value
+        return name, value
+
     def filter_by_target_object(self, queryset, name, value):
         """Filter that returns the tags associated with target."""
-        TARGET_TYPES = {
+        target_types = {
             "course_id": "courseoverview",
             "username": "user",
+            "certificate_id": "generatedcertificate",
+            "certificate_key": "generatedcertificate",
+            "certificate_url": "generatedcertificate",
         }
         if value:
-            DEFAULT = {
-                name: value,
+            model_field_name, model_value = self._map_to_model(name, value)
```

Comment:
This call to the _map_to_model function looks like it is trying to abstract some logic that should just be explicit.

Reply:
I can change it to `parse_search_string`

Reply:
We have an explicit mapping donde vía the `target_types` dict and a implicit mapping done via the `_map_to_model` function that looks at the string and returns `int(value)` for "certificate_id" and or `value` otherwise.

Thats is really convoluted. We should go for one of the two. Either a function that returns the `filter_params` for different all types vía a long mapping dictionary, or something that does different functions for each type of target_type and we call each function by looking at the target_type and calling one for course_id, other for username and other for certificates, ...

Reply:
I had already published the other comment when I noticed yours @mariajgrimaldi.

I don't mean the function name. I mean the actual function. We either do it implicitly (a function that does `if something then`) or explicitly (a big dict with all the mapping in it) but doing one for some types of targets and something different for certificates is plain bad practice.